skopt/dummy_opt.py

```diff
@@ -52,14 +70,32 @@ def dummy_minimize(func, dimensions, n_calls=100, random_state=None):
     """
     rng = check_random_state(random_state)
     space = Space(dimensions)
-    X = space.rvs(n_samples=n_calls, random_state=rng)
-
-    init_y = func(X[0])
+    if x0 is None:
+        x0 = []
+    x0 = list(x0)
+    if x0 and not isinstance(x0[0], list):
+        x0 = [x0]
+    if y0 is None:
+        y0 = [func(x) for x in x0]
+        n_calls -= len(y0)
+    if isinstance(y0, Iterable):
+        y0 = list(y0)
+    else:
+        y0 = [y0]
+    if len(x0) != len(y0):
+        raise ValueError("x0 and y0 should have the same length")
+    X = x0 + space.rvs(n_samples=n_calls, random_state=rng)
+    init_provided = len(y0) != 0
```

Comment:
I'm not sure what you mean, if `y0` is provided, you are making the checks here right (https://github.com/scikit-optimize/scikit-optimize/pull/119/files/98df84c12a8d13b5f62393b16357d7ee54d864c0#r71977002)?